agate-webapp/src/main/java/org/obiba/agate/web/filter/auth/oidc/AgateCallbackFilter.java

```diff
@@ -366,4 +352,15 @@ public Wrapper() {
     }
   }
 
+  private void sendRedirectOrSendError(String redirect, String message, HttpServletResponse response) throws IOException {
+    if (Strings.isNullOrEmpty(redirect)) {
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, message);
```

Comment:
Is it a internal server error or a bad request error?

Reply:
I'd say it's an internal server error because the error occurs in the callback, I don't think bad request would qualify it.